### PR TITLE
CorelliPowderCalibrationCreate accepts MatrixWorkspaces

### PR DESCRIPTION
**Pull request by jmborr**
**Description of work.**
- [x] Algorithm accepts matrix workspaces
- [x] utility function to load long runs
- [x]  no need to update the release notes because the algorithm is absent in the current stable release.
- [x] manual test

**To test:**
Run the following script

```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from corelli.calibration.powder import load_and_rebin


Load(Filename='CORELLI_124036-124037', BankName='bank55,bank80', OutputWorkspace='events', LoadLogs=False)
CorelliPowderCalibrationCreate(InputWorkspace='events',
                               OutputWorkspacesPrefix='events_',
                               TofBinning=[3000,-0.001,16660],
                               PeakPositions=[1.3143, 1.3854,1.6967, 1.8587, 2.0781, 2.3995, 2.9388, 4.1561],
                               ComponentList='bank55/sixteenpack,bank80/sixteenpack')
                               
load_and_rebin(runs=[124036, 124037], output_workspace='counts', rebin_params=[3000,-0.0001,16660], banks=[55, 80])
CorelliPowderCalibrationCreate(InputWorkspace='counts',
                               OutputWorkspacesPrefix='counts_',
                               TofBinning=[3000,-0.001,16660],
                               PeakPositions=[1.3143, 1.3854,1.6967, 1.8587, 2.0781, 2.3995, 2.9388, 4.1561],
                               ComponentList='bank55/sixteenpack,bank80/sixteenpack')
```
Although the tables  `events_adjustments` and `counts_adjustments` are not identical, both sets of adjustments produce a distribution  of peak deviations very similar.  Verify you obtain the below plot by plotting the same spectrum numbers for workspaces `events_percent_peak_deviations_adjusted`, `events_percent_peak_deviations_adjusted`, and `counts_percent_peak_deviations_adjusted`

Capture

Refs #30250

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KedoKudo** [comment]
Review process

- [ ]  branch build on analysis cluster
- [ ]  testing scripts pass
- [ ]  going through src changes

**Comment by jmborr** [comment]
@KedoKudo I've never built Mantid in the analysis cluster. Do you have a script to that effect I could use?

**Comment by KedoKudo** [comment]
> @KedoKudo I've never built Mantid in the analysis cluster. Do you have a script to that effect I could use?

Check the `makefile` in this repo :smile: 

https://github.com/KedoKudo/MantidWorkbench_dev_template

You do need to switch to gcc-7 before start the building, which can be done using `scl enable devtoolset-7 bash`.
Also, you might want to turn off your conda env so that cmake can use the system Python directly.

**Comment by jmborr** [comment]
why is the command `scl enable devtoolset-7 bash` not part of your `Makefile` ?

**Comment by KedoKudo** [comment]
> why is the command `scl enable devtoolset-7 bash` not part of your `Makefile` ?

`gmake` executes each command in a subprocess, so there is no easy way to change the overall env settings from a makefile.

File: scripts/corelli/calibration/powder.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        single_run = '__single_run_' + output_workspace
+        for i, run in enumerate(runs[1:]):
+            logger.information(f'Loading run {run}. {len(runs) - 1 - i} runs remaining to be loaded')
+            LoadEventNexus(Filename=f'{instrument}_{run}', OutputWorkspace=single_run, LoadLogs=False, **kwargs)
+            if rebin_params is not None:
+                Rebin(InputWorkspace=single_run, OutputWorkspace=single_run, Params=rebin_params, PreserveEvents=False)
+            Plus(LHSWorkspace=output_workspace, RHSWorkspace=single_run, OutputWorkspace=output_workspace)
+            DeleteWorkspace(single_run)  # save memory as quick as possible
+    except RuntimeError:
+        DeleteWorkspace(single_run)  # a bit of clean-up
```
Comment (KedoKudo) [inline scripts/corelli/calibration/powder.py:53]: According to the official documentation for Python3 (https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions), it is recommended to use `try-except-finally` for consistent cleaning up.

> this is just a FYI, not a request for change :wink:

**Review by KedoKudo — APPROVED** [review]
This PR pass building on analysis cluster, and the provided test scripts produces the calibration table as expected.

**Pull request merged**